Source/ablastr/fields/MultiFabRegister.H

```diff
@@ -0,0 +1,697 @@
+/* Copyright 2024 The ABLASTR Community
+ *
+ * This file is part of ABLASTR.
+ *
+ * License: BSD-3-Clause-LBNL
+ * Authors: Axel Huebl
+ */
+#ifndef ABLASTR_FIELDS_MF_REGISTER_H
+#define ABLASTR_FIELDS_MF_REGISTER_H
+
+#include <AMReX_BoxArray.H>
+#include <AMReX_DistributionMapping.H>
+#include <AMReX_Enum.H>
+#include <AMReX_Extension.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_MultiFab.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Vector.H>
+
+#include <array>
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+
+namespace
+{
+    // type trait helpers in lieu of an amrex::is_amrex_enum
+    template <typename, typename = std::void_t<>>
+    struct is_castable_to_string : std::false_type {};
+
+    template <typename T>
+    struct is_castable_to_string<T, std::void_t<decltype(static_cast<std::string>(std::declval<T>()))>> : std::true_type {};
+
+    /** helper to either cast a string/char array to string to query an AMREX_ENUM */
+    template<typename T>
+    std::string getExtractedName (T name)
+    {
+        if constexpr(is_castable_to_string<T>())
+        {
+            // already a unique string key
+            return std::string(name);
+        } else
+        {
+            // user-defined AMREX_ENUM or compile error
+            return amrex::getEnumNameString(name);
+        }
+    }
+}
+
+namespace ablastr::fields
+{
+    /** Components (base vector directions) of vector/tensor fields.
+     *
+     * Because of different staggering, the components of vector/tensor fields are stored
+     * in separate (i)MultiFab.
+     *
+     * @todo: synchronize with AMReX "enum class Direction"
+     */
+    struct Direction
+    {
+        int dir = 0;
+
+        bool operator<(const Direction& other) const
+        {
+            return other.dir < this->dir;
+        }
+
+        /* TODO: just temporary int compatibility */
+        operator int() const { return dir; }
+    };
```

Comment:
Since there is a `Direction` available in AMReX, which should serve the same purpose, maybe we can try to use that one and see if it works for us. AMReX's direction is defined [here](https://github.com/AMReX-Codes/amrex/blob/1bfdb84c124a6feb6b19df976b6e7ceb4e6c016e/Src/Base/AMReX_Orientation.H#L14):
```cpp
enum class Direction : int { AMREX_D_DECL(x = 0, y = 1, z = 2) };
```

This could be another separate follow-up issue/PR.

Reply:
Yes, see the `@todo` in the comment above.

We currently need to add some implicit casts, thus we cannot use it.